tiledb/sm/misc/comparators.h

```diff
@@ -149,75 +162,60 @@ class GlobalCmp {
    * @param b The second coordinate.
    * @return `true` if `a` precedes `b` and `false` otherwise.
    */
-  bool operator()(const ResultCoords<T>& a, const ResultCoords<T>& b) const {
+  bool operator()(const ResultCoords& a, const ResultCoords& b) const {
     // Compare tile order first
-    auto tile_cmp =
-        domain_->tile_order_cmp_tile_coords<T>(a.tile_coords_, b.tile_coords_);
+    if (tile_order_ == Layout::ROW_MAJOR) {
+      for (unsigned d = 0; d < dim_num_; ++d) {
+        auto res = domain_->tile_order_cmp(d, a.coord(d), b.coord(d));
 
-    if (tile_cmp == -1)
-      return true;
-    if (tile_cmp == 1)
-      return false;
-    // else tile_cmp == 0 --> continue
+        if (res == -1)
+          return true;
+        if (res == 1)
+          return false;
+        // else same tile on dimension d --> continue
+      }
+    } else {  // COL_MAJOR
```

Comment:
for these scenarios, i like to put something like `assert(tile_order_ == Layout::COL_MAJOR)`, just to help protect against a regression if the enum expands. Unlikely in this scenario, though.